api/base/utils.py

```diff
@@ -137,3 +137,9 @@ def default_node_permission_query(user):
 
 def extend_querystring_params(url, params):
     return furl.furl(url).add(args=params).url
+
+def extract_object_from_dict(self, obj):
```

Comment:
This is not immediately clear why it's doing what it is, so please add a docstring.